adapter/internal/adapter/adapter.go

```diff
@@ -234,6 +232,7 @@ func Run(conf *config.Config) {
 		synchronizer.FetchKeyManagersOnStartUp(conf)
 		go synchronizer.UpdateKeyTemplates()
 		go synchronizer.UpdateBlockingConditions()
+		messaging.ProcessEvents(conf)
```

Comment:
Is there a specific reason for remove this running as a go routine

Reply:
we are waiting till all list variables are initiated so that we dont get any nil pointers when events are populated.